File: code_examples/3-Structured_Problem_Solving/8-If_Then_ElseIf_Else/Java/Main.java
/*
 * This program checks a traffic light
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // this function checks a traffic light
        String lightColor;

        // input
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the color of the traffic light: ");
        lightColor = scanner.nextLine();

        // process and output
        if (lightColor.equals("red")) {
            System.out.println("Stop!");
        } else if (lightColor.equals("yellow")) {
            System.out.println("Slow down!");
        } else if (lightColor.equals("green")) {
            System.out.println("Go!");
        } else {
            System.out.println("Not a valid color.");
        }

        System.out.println("\nDone.");
    }
}

File: code_examples/3-Structured_Problem_Solving/8-If_Then_ElseIf_Else/CPP/main.cpp
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program checks if ther is over 30 students

#include <iostream>

int main() {
    // this function checks if ther is over 30 students
    const int MAX_STUDENT_NUMBER = 30;
    int numberOfStudents;

    // input
    std::cout << "Enter the number of students: ";
    std::cin >> numberOfStudents;

    // process and output
    if (numberOfStudents > MAX_STUDENT_NUMBER) {
        std::cout << "Too many students!\n";
    } else {
        std::cout << "Not too many students.\n";
    }

    std::cout << "\nDone." << std::endl;
}
